src/core/api/api_access_control.pl

```diff
@@ -0,0 +1,486 @@
+                   System_Context),
+
+    with_transaction(
+        System_Context,
+        replace_document(System_Context,Typed_Role,_),
+        _
+    ).
+
+api_delete_role(_, Auth, Role_Id) :-
+    do_or_die(
+        is_super_user(Auth),
+        error(access_not_authorised(Auth,'Action/manage_capabilities','SystemDatabase'), _)),
+
+    create_context(system_descriptor{}, commit_info{author: "admin", message: "API: Update Role"},
+                   System_Context),
+
+    % Make sure we are actually dealing with a role document
+    prefix_expand(Role_Id, _{ '@base' : 'terminusdb://system/data/'
+                            }, Expanded_Role_Id),
+    do_or_die(
+        get_document_uri_by_type(System_Context, 'http://terminusdb.com/schema/system#Role',
+                                 Expanded_Role_Id),
+        error(document_not_found(Role_Id), _)
+    ),
+
+    with_transaction(
+        System_Context,
+        delete_document(System_Context,Role_Id),
+        _
+    ).
+
+api_get_organizations(SystemDB, Auth, Organizations) :-
+    do_or_die(
+        is_super_user(Auth),
+        error(access_not_authorised(Auth,'Action/manage_capabilities','SystemDatabase'), _)),
+    get_organizations(SystemDB,Organizations).
+
+get_organizations(SystemDB, Organizations) :-
+    findall(Organization,
+            ask(SystemDB,
+                (   t(Id,rdf:type,'@schema':'Organization'),
+                    get_document(Id, Organization))),
+            Organizations).
+
+api_get_organization_from_name(SystemDB, Auth, Name, Organization) :-
+    do_or_die(
+        is_super_user(Auth),
+        error(access_not_authorised(Auth,'Action/manage_capabilities','SystemDatabase'), _)),
+    do_or_die(
+        get_organization_from_name(SystemDB, Name, Organization),
+        error(no_id_for_organization_name(Name), _)).
+
+get_organization_from_name(SystemDB, Name, Organization) :-
+    ask(SystemDB,
+        (   t(Id,name,Name^^xsd:string),
+            t(Id,rdf:type,'@schema':'Organization'),
+            get_document(Id,Organization))).
+
+api_get_organization_from_id(SystemDB, Auth, Id, Organization) :-
+    do_or_die(
+        is_super_user(Auth),
+        error(access_not_authorised(Auth,'Action/manage_capabilities','SystemDatabase'), _)),
+    do_or_die(
+        get_organization_from_id(SystemDB, Id, Organization),
+        error(no_organization_with_given_id(Id), _)).
+
+get_organization_from_id(SystemDB, Id, Organization) :-
+    ask(SystemDB,
+        (   t(Id,rdf:type,'@schema':'Organization'),
+            get_document(Id,Organization))).
+
+api_add_organization(_, Auth, Organization, Id) :-
+    do_or_die(
+        is_super_user(Auth),
+        error(access_not_authorised(Auth,'Action/manage_capabilities','SystemDatabase'), _)),
+
+    % Make sure that we only insert Organization types.
+    put_dict('@type', Organization, 'Organization', Organization1),
```

Comment:
I don't think it's a good idea to add Organization into the system database after making slight changes to the document. While you blacklist the database field here, you're not blacklisting whatever other fields we may come up with in the future, and it'll be hard in future reviews to remember that there's this place in the code that needs to be changed to make sure people don't just insert pre-filled records into our database.

Better to start with a fresh document and fill it in, rather than take things out.